Automattic/jetpack
# [RNMobile] Fix infinite loop error in unit test

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4168

`Gutenberg-Mobile:` https://github.com/wordpress-mobile/gutenberg-mobile/pull/4263

As described in the above issue, [the `inserts the block` test](https://github.com/wordpress-mobile/gutenberg-mobile/blob/a6155a45fc3d5d9417d36efa48b9824d73476108/src/jetpack/test/tiled-gallery.js#L48) was failing to run with the following error:

```
Error: Uncaught [Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.]
```

The logs pointed to [this `setMaxWidth` call](https://github.com/Automattic/jetpack/blob/fe69bde8661f558b26ce4f61408f73e3f38775f9/projects/plugins/jetpack/extensions/blocks/tiled-gallery/edit.native.js#L55) in the block's edit function. 

#### Changes proposed in this Pull Request:

* An infinite loop appears to have been caused by the fact `setMaxWidth` updates the `sizes` variable, which was called in the `useEffect` hook's dependency array. 
* In this PR, I've removed `sizes` from the hook's dependency array and replaced it with `width`, which I believe achieves the same effect. (`height` is the only other item in the `sizes` object, and it doesn't seem necessary to track its changes for the functionality in this `useEffect` hook.)

#### Jetpack product discussion

p9ugOq-1Tb-p2 

#### Does this pull request change what data or activity we track or use?

No, it doesn't.

#### Testing instructions:

To test, it'll be necessary to first remove the `.skip` command added [here](https://github.com/wordpress-mobile/gutenberg-mobile/blob/0e1c9bae48d83308fa9d4e2c2aaae6cf1af3440a/src/jetpack/test/tiled-gallery.js#L47) then run `npm run test src/jetpack/test/tiled-gallery.js` while cd'd into the gutenberg-mobile folder from the terminal.

With the changes from this PR applied, the test should now run past the block creation step.

`.skip` isn't removed with this PR as the test still fails at [the `expectedHtml ` step](https://github.com/wordpress-mobile/gutenberg-mobile/blob/a6155a45fc3d5d9417d36efa48b9824d73476108/src/jetpack/test/tiled-gallery.js#L81) due to the issue described at https://github.com/wordpress-mobile/gutenberg-mobile/issues/4262, which will be addressed in a separate PR.
